src/systems/systems.component.tsx

```diff
@@ -508,11 +515,8 @@ function Systems() {
       <SystemDialog
         open={menuDialogType !== undefined && menuDialogType !== 'delete'}
         onClose={() => setMenuDialogType(undefined)}
-        type={
-          menuDialogType !== undefined && menuDialogType !== 'delete'
-            ? menuDialogType
-            : 'edit'
-        }
+        requestType={getRequestType()}
```

Comment:
```suggestion
        requestType={menuDialogType === 'edit' ? 'patch' : 'post'}
```
Not sure its worth having a function inside the component body - it will just get redefined each render. I think either this, or having it outside the function taking in menuDialogType is preferable.